src/lib/pages/backoffice/Document/DocumentDetails/DocumentBorrowingRequests/DocumentBorrowingRequests.js

```diff
@@ -0,0 +1,172 @@
+      hasError: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchBorrowingRequests();
+  }
+
+  componentWillUnmount() {
+    this.cancellableBorrowingRequests &&
+      this.cancellableBorrowingRequests.cancel();
+  }
+
+  currentBorrowingRequestsQuery = () => {
+    const {
+      documentDetails: { pid },
+    } = this.props;
+    const illConfig = invenioConfig.ILL_BORROWING_REQUESTS;
+    const statuses = _difference(
+      illConfig.orderedValidStatuses,
+      illConfig.completedStatuses
+    );
+    return borrowingRequestApi.list(
+      borrowingRequestApi.query().withDocPid(pid).withState(statuses).qs()
+    );
+  };
+
+  fetchBorrowingRequests = async () => {
+    try {
+      this.cancellableBorrowingRequests = withCancel(
+        this.currentBorrowingRequestsQuery()
+      );
+      const response = await this.cancellableBorrowingRequests.promise;
+      this.setState({
+        borrowingRequests: response.data,
+        hasError: false,
+        isLoading: false,
+      });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({ hasError: true, isLoading: false });
+      }
+    }
+  };
+
+  seeAllButton = () => {
+    const path = ILLRoutes.borrowingRequestListWithQuery(
+      this.currentBorrowingRequestsQuery()
+    );
+    return <SeeAllButton to={path} />;
+  };
+
+  viewDetails = ({ row }) => {
+    return (
+      <Link to={ILLRoutes.borrowingRequestDetailsFor(row.id)}>{row.id}</Link>
+    );
+  };
+
+  renderTable(data) {
+    const { showMaxBorrowingRequests } = this.props;
+    const columns = [
+      {
+        title: 'Loan',
+        formatter: this.viewDetails,
+      },
+      {
+        title: 'Provider',
+        formatter: 'metadata.provider.name',
+      },
+      {
+        title: 'Created on',
+        field: 'created',
+        formatter: dateFormatter,
+      },
+      {
+        title: 'State',
+        field: 'metadata.status',
+      },
+      {
+        title: 'Requested on',
+        field: 'metadata.request_date',
+        formatter: dateFormatter,
+      },
+      {
+        title: 'Expected delivery',
+        field: 'metadata.expected_delivery_date',
+        formatter: dateFormatter,
+      },
+      {
+        title: 'Received on',
+        field: 'metadata.received_date',
+        formatter: dateFormatter,
+      },
+    ];
+
+    return (
+      <ResultsTable
+        data={data.hits}
+        columns={columns}
+        totalHitsCount={data.total}
+        name="borrowingRequests"
+        seeAllComponent={this.seeAllButton()}
+        showMaxRows={showMaxBorrowingRequests}
+      />
+    );
+  }
+
+  renderResultsOrEmpty(data) {
+    return data.total > 0 ? (
+      this.renderTable(data)
+    ) : (
+      <Message info icon data-test="no-results">
+        <Message.Content>
+          <Message.Header>No results found</Message.Header>
+          There are no ongoing interlibrary loans.
+        </Message.Content>
+      </Message>
+    );
+  }
```

Comment:
I think the ResultsTable component has this part backed in already, to be confirmed